### PR TITLE
ToAU / Instanced Battle fixes

### PR DESCRIPTION
**Pull request by ymkc**

### CODE REVIEW

File: scripts/zones/Alzadaal_Undersea_Ruins/npcs/_20m.lua [hunk 1]
```diff
@@ -106,12 +106,12 @@ function onEventUpdate(player,csid,option,target)
             for i,v in ipairs(party) do
                 if (v:getID() ~= player:getID()) then
                     if (v:hasKeyItem(NYZUL_ISLE_ROUTE) == false and v:hasCompletedMission(TOAU, PATH_OF_DARKNESS) == false) then
-                        player:messageText(target,MEMBER_NO_REQS, false);
-                        player:instanceEntry(target,1);
+                        v:messageText(target,MEMBER_NO_REQS, false);
```
Comment (teschnei) [inline scripts/zones/Alzadaal_Undersea_Ruins/npcs/_20m.lua]: these changes aren't right - it should inform the player trying to register the instance when a party member doesn't meet the requirements, not the party member

Comment (teschnei) [reply scripts/zones/Alzadaal_Undersea_Ruins/npcs/_20m.lua]: (you left all the other ones correct though, just this one is wrong)

File: scripts/globals/allyassist.lua [hunk 2]
```diff
@@ -67,6 +67,10 @@ function startAllyAssist(entity, assistMode)
 
     local target = targetMobs[math.random(#targetMobs)]
 
+    if (target == null) then
```
Comment (TeoTwawki) [inline scripts/globals/allyassist.lua]: lua uses "nil" not null.

**Comment by teschnei** [comment]
Needed fixes before merge

**Pull request merged**